ryan_savill/03_background_subtraction/readme.md

```diff
@@ -0,0 +1,281 @@
+- And believe it or not that was it! 
+
+The important thing is that this python file is in the same folder as the python file you want to use your library in. If this is the case you should be able to import it like this:
+
+
+```python
+import img_analysis_functs as iaf
+
+io.imshow(iaf.subtract_background(tribolium))
+```
+
+
+    
+![png](output_17_2.png)
+    
+
+
+## Difference of Gaussian
+This is another method of highlighting nuclei, even if it is not described as background subtraction but rather [feature enhancement](https://en.wikipedia.org/wiki/Difference_of_Gaussians). In essence what is done is that we generate two gaussian blurred images with differing sizes of gaussians. The features we want to highlight should be in between the two sizes of the gaussians. Afterwards we subtract the larger blurred image from the smallller one. In a way this combines the noise removal of the small gaussian blurred image with a background subtraction represented by the larger gaussian blurred image. If filters are a common concept for you you can think of it as a bandpass filter that highlights thefeatures in between the two sizes. Since gaussian blur is a simple function included in scikit image let's create this filter ourselves:
+
+
+```python
+from skimage import filters
+
+# the smaller gaussian is just there to remove 
+# noise so it can be just a few pixels
+small_gauss = filters.gaussian(tribolium, sigma = 1)
+
+# the larger gaussian has to be bigger than our largest
+# object, which we can measure or make an educated guess
+large_gauss = filters.gaussian(tribolium, sigma = 100)
+
+# now we subtract the large gaussian from the small one
+dog_tribolium = small_gauss - large_gauss
+
+# visualising the results
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize =(20,20))
+ax1.imshow(small_gauss, cmap = 'gray')
+ax1.set_title('Small Gaussian')
+ax2.imshow(large_gauss, cmap = 'gray')
+ax2.set_title('Large Gaussian')
+ax3.imshow(dog_tribolium, cmap = 'gray')
+ax3.set_title('DoG Image')
+```
+
+
+    
+![png](output_19_1.png)
+    
+
+
+We can see that it is not quite as effective as the tophat-filter but nonetheless, it has managed to remove some of the background in the image. Making this filter ourselves is a good exercise in coding and understanding what these filters do but when it gets to complicated filters reinventing the wheel isn't practical and you will most likely find the filter you are looking for either in [scikit image](https://scikit-image.org/), in [scipy](https://docs.scipy.org/doc/) or just google (this is what I often do). Because python is a relatively simple language to learn it has many users who have most likely asked the same question as you have countless times! For the difference of gaussian filter scikit image has this function, which we can use here instead of doing the processing ourselves:
```

Comment:
```suggestion
We can see that it is not quite as effective as the top-hat filter but nonetheless, it has managed to remove some of the background in the image. Making this filter ourselves is a good exercise in coding and understanding what these filters do. However, when it gets to complicated filters, reinventing the wheel isn't practical and you will most likely find the filter you are looking for either in [scikit image](https://scikit-image.org/), in [scipy](https://docs.scipy.org/doc/) or just google (this is what I often do). Because python is a relatively simple language to learn. There are many users who have most likely asked the same question as you have! For the [Difference of Gaussian filter](https://scikit-image.org/docs/dev/api/skimage.filters.html#skimage.filters.difference_of_gaussians) scikit image has this function as well, which we can use here instead of doing the processing ourselves:
```
Wording + links